File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] solution(int k, int[] score) {
        List<int> answer = new List<int>();
        List<int> best = new List<int>();
        
        foreach(var s in score)
        {
            best.Add(s);
            best.Sort();
            if(best.Count > k)
            {
                best.RemoveAt(0);
            }
            answer.Add(best[0]);
        }
        return answer.ToArray();
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(int k, vector<int> score) {
    vector<int> answer;
    priority_queue<int, vector<int>, greater<int>> pq;
    for(int n : score)
    {
        if(pq.size() < k)
        {
            pq.push(n);
        }
        else
        {
            if(pq.top() < n)
            {
                pq.pop();
                pq.push(n);
            }
        }
        answer.push_back(pq.top());
    }
    return answer;
}